### PR TITLE
Properly use mag detections for mag photometry boundary lines

### LINKED ISSUE #996: First/last detection lines on photometry plot sometimes get assigned to non-detections
E.g., ZTF20acdnqzm:

https://private.caltech.edu/source/ZTF20acdnqzm

The first detection vertical line is being drawn over a non-detection. 

Screen Shot 2020-09-26 at 7 28 34 PM

### PR DESCRIPTION
**Pull request by kmshin1397**
Closes #996

### CODE REVIEW

File: skyportal/plot.py [hunk 1]
```diff
@@ -454,7 +454,13 @@ def photometry_plot(obj_id, user, width=600, height=300):
     )
 
     # Mark the first and last detections again
+    data['hasmag'] = ~data['mag'].isna()
+    detection_dates = data[data['hasmag']]['mjd']
```
Comment (dannygoldstein) [inline skyportal/plot.py:458]:
```suggestion
    detection_dates = data[obsind]['mjd']
```

File: skyportal/plot.py [hunk 2]
```diff
@@ -454,7 +454,13 @@ def photometry_plot(obj_id, user, width=600, height=300):
...
     if len(detection_dates) > 0:
+        first = round(detection_dates.min(), 6)
+        last = round(detection_dates.max(), 6)
+        first_x = np.full(5000, first)
+        last_x = np.full(5000, last)
```
Comment (dannygoldstein) [inline skyportal/plot.py:463]: Are these used anywhere?

Comment (kmshin1397) [reply skyportal/plot.py]: They were, but only once so I just moved these to be inline

File: skyportal/plot.py (hunk 1 above)
Comment (kmshin1397) [reply skyportal/plot.py]: Done

**Comment by kmshin1397** [comment]
@dannygoldstein Feedback addressed

**Pull request merged**